lib/vanity/adapters/active_record_adapter.rb

```diff
@@ -20,7 +20,7 @@ def self.needs_attr_accessible?
 
         def self.rails_agnostic_find_or_create_by(method, value)
           if respond_to? :find_or_create_by
-            find_or_create_by(method => value)
+            find_or_create_by_without_protection(method => value)
```

Comment:
I don't think this is the right approach - this will basically nullify the use of protected attributes. If you could post the stacktrace of the error you were seeing it'd help to track down what was causing the issue.

Looking at VanityExperiment, we might need to add something like `attr_accessible :metric_id if needs_attr_accessible?` to the VanityMetric class, but both of these are basically primary keys and probably shouldn't be mass assignable.